### ISSUE TITLE
Parse error: syntax error, unexpected ')', expecting '&' or T_VARIABLE

### ISSUE THREAD

**Issue by RowboTony** [teampoop/commerce_stripe#1]
Hi, I tried to enable this module and got the following error:

`Parse error: syntax error, unexpected ')', expecting '&' or T_VARIABLE in /var/www/commerce-1.3/sites/default/modules/commerce_stripe/commerce_stripe.module on line 184
Drush command terminated abnormally due to an unrecoverable error.                                      [error]
Error: syntax error, unexpected ')', expecting '&' or T_VARIABLE in /var/www/commerce-1.3/sites/default/modules/commerce_stripe/commerce_stripe.module, line 184`

I was having issues with the _other_ commerce_stripe module on drupal.org, http://drupal.org/project/commerce_stripe, I've used other teampoop modules and know they are solid and stable. I'm hoping we can get this one working to get my new Drupal Commerce store online with Stripe payments.

Thank you!
--Tony @drupleg

**Comment by teampoop** [comment]
Hi ,

 I will double check on that error. What URL path were you at when you
got that error? I kinda bailed on this version and helped with the
other one.

On Jun 23, 2012, at 11:39 PM, tgroff
reply@reply.github.com
wrote:

**Comment by RowboTony** [comment]
Hi, thanks for the quick reply! Realizing that's your code in the d.o module it's probably best for me to stay in that issue queue to work things out  (see: http://drupal.org/node/1655498). For the sake of completeness here's some more details on what I experienced with this git module. Thanks Markie! 

Are you using the d.o commerce_stripe module? Are you having any problems with it? I'm also quite willing to pay for assistance with this as my store can't launch live until I figure out what's causing the integer and incorrect charges with the other module.  Thank you for your time and help.

I did a little more poking around comparing this commerce_stripe module with the one on d.o, and I found much of the code is very similar.  I got the original error with this git module when running #drush en commerce_stripe, the module didn't enable and showed the error `arse error: syntax error, unexpected ')', expecting '&' or T_VARIABLE`.  It looks like the last line of code is incomplete/missing `{}`? So I deleted that line and then enabled this commerce_stripe module again, didn't get the error this time on enable (also disabled and deleted the other module, and clearing caches before enabling this module).

Deleted these lines:

```
function _commerce_stripe_set_transaction($payment_id, $remote_id, $amount, )
```

Then I went to `admin/commerce/config/payment-methods/manage/commerce_payment_commerce_stripe/edit/3`, input my keys, clicked (save) and it all blowed up with a 500 internal server error... looks like that last line was probably important :) Here's the apache log error:

```
[24-Jun-2012 10:15:57] PHP Fatal error:  Cannot use object of type RulesAction as array in /var/www/commerce-1.3/sites/default/modules/commerce_stripe/commerce_stripe.module on line 141
```

Thanks again for the response... I'm kind at a loss here as I am not a programmer, I'm a sitebuilder and server admin so I can fake it most of the time :).  I'm just completely baffled as to why the d.o module is misbehaving and not charging the correct amounts (see: http://drupal.org/node/1655498), so I figured to try this one to see if it worked.  

--Tony
